src/does-match.js

```diff
@@ -37,45 +37,108 @@
       return string;
     }
 
-    var replaced = '';
-    for (var i = 0, size = string.length; i < size; i++) {
-      replaced += replaceDiacritics(string[i]);
-    }
+    var replaced = string.split('').map(replaceDiacritics);
 
-    return replaced;
+    return replaced.join('');
```

Comment:
@ehkasper I also prefer what you did here.

I'm just a little bit worried about a possible performance impact, since `does-match` will usually be used inside some loop that might be somewhat big.

It would be nice to have some insight about how this change would affect performance (for better, for worse, or no significant change at all). I'll try to make some benchmark, would appreciate if you could bring some results too.